hack/generated/controllers/generic_controller.go

```diff
@@ -164,9 +192,27 @@ func register(mgr ctrl.Manager, applier armclient.Applier, obj runtime.Object, l
 
 	reconciler.Controller = c
 
-	return ctrl.NewWebhookManagedBy(mgr).
-		For(obj).
-		Complete()
+	return nil
+}
+
+// MakeGroupVersionMap creates a map of group to version. This returns an error if there is more than one group to version mapping
+// for a given group.
+func MakeGroupVersionMap(mgr ctrl.Manager, objs []runtime.Object) (map[string]schema.GroupVersion, error) {
```

Comment:
This pattern won't work if we have a situation where a group has some resources which were removed in an old version, as then the storage type is that old version. I should change this to be based on Kind -- you can look up the Group-Version for each Kind.

Reply:
Do you mean group-kind? We're modelling the different RPs as different groups, right?

Reply:
Yeah I mean group-kind